src/__tests__/app-monitor.test.ts

```diff
@@ -0,0 +1,171 @@
+  })
+
+  /**
+   * Role
+   */
+
+  const identityPoolLogicalId = appMonitorStack.getLogicalId(
+    appMonitor.identityPool,
+  )
+
+  const roleCfnElement = appMonitor.guestRole.node.findChild("Resource")
+  const roleLogicalId = appMonitorStack.getLogicalId(
+    roleCfnElement as CfnElement,
+  )
+
+  template.hasResourceProperties("AWS::IAM::Role", {
+    AssumeRolePolicyDocument: {
+      Statement: [
+        Match.objectEquals({
+          Action: "sts:AssumeRoleWithWebIdentity",
+          Effect: "Allow",
+          Condition: {
+            StringEquals: {
+              "cognito-identity.amazonaws.com:aud": {
+                Ref: identityPoolLogicalId,
+              },
+            },
+            "ForAnyValue:StringEquals": {
+              "cognito-identity.amazonaws.com:amr": "unauthenticated",
+            },
+          },
+          Principal: { Federated: "cognito-identity.amazonaws.com" },
+        }),
+      ],
+    },
+  })
+
+  template.resourceCountIs("AWS::IAM::Role", 1)
+
+  /**
+   * Policy for put events
+   */
+  template.hasResourceProperties("AWS::IAM::Policy", {
+    PolicyDocument: {
+      Statement: Match.arrayEquals([
+        Match.objectLike({
+          Action: "rum:PutRumEvents",
+          Effect: "Allow",
+          Resource: {
+            "Fn::Join": [
+              "",
+              [
+                "arn:",
+                { Ref: "AWS::Partition" },
+                ":rum:",
+                { Ref: "AWS::Region" },
+                ":",
+                { Ref: "AWS::AccountId" },
+                `:appmonitor/${appMonitor.appMonitor.name}`,
+              ],
+            ],
+          },
+        }),
+      ]),
+    },
+    Roles: [Match.objectLike({ Ref: roleLogicalId })],
+  })
+
+  template.resourceCountIs("AWS::IAM::Policy", 1)
+
+  /**
+   * Role attachment in order to have unathenticated users assume role so put:events is allowed
+   */
+  template.hasResourceProperties("AWS::Cognito::IdentityPoolRoleAttachment", {
+    IdentityPoolId: { Ref: identityPoolLogicalId },
+    Roles: {
+      unauthenticated: {
+        "Fn::GetAtt": Match.arrayEquals([roleLogicalId, "Arn"]),
+      },
+    },
+  })
+
+  template.resourceCountIs("AWS::Cognito::IdentityPoolRoleAttachment", 1)
+})
+
+test("App monitor has default alarm for high volume events", () => {
+  const app = new App()
+
+  const appMonitorStack = new Stack(app, "AppMonitorStack")
+
+  const alarmActionTopic = new sns.Topic(appMonitorStack, "Topic")
+
+  const alarmAction = new cw_actions.SnsAction(alarmActionTopic)
+  new WebappMonitor(appMonitorStack, "WebappMonitor", {
+    webappDomainName: "webappdomain.example",
+    appMonitorVersion: "v1",
+    telemetries: ["http", "errors", "performance"],
+    eventsAlarmAction: {
+      alarmAction: alarmAction,
+      okAction: alarmAction,
+    },
+  })
+
+  const template = Template.fromStack(appMonitorStack)
+
+  const snsTopicLogicalId = appMonitorStack.getLogicalId(
```

Comment:
An alternative way of doing this:
```suggestion
  const snsTopicLogicalId = (snsTopic.node.defaultChild as CfnElement).logicalId
  // OR: const snsTopicLogicalId = (snsTopic.node.defaultChild as sns.CfnTopic).ref
```

Reply:
neither of these worked. 

I think it might be due to the scope? Not certain why it's necessary to create the stack "appMontiorStack" in tests, but thinking it has something to do with this?

Reply:
It's probably because they are CDK tokens. `appMonitorStack.getLogicalId(alarmActionTopic.node.defaultChild as CfnResource)` is another alternative that is a bit more concise that I think should work, but your approach is fine 👍